tests/test_cursors.py

```diff
@@ -158,6 +158,19 @@ def foo(n: size, m: size):
     assert str(foo1.forward(stmt)._impl._node) == golden
 
 
+def test_simplify_predicates_forwarding():
+    @proc
+    def foo(n: size):
+        assert n >= 1
```

Comment:
If I understand correctly, this PR is fixing the forwarding of simplify when asserts get simplified, right? Is `n >= 1` gonna get simplified even further? My question is whether this test is actually testing the condition `if new_preds: ...`

Reply:
`simplify` modifies all expressions, regardless of whether their actual contents actually change. 

The issue before was that `simplify` resulted in a new `proc` due to changes to the predicates, but the forwarding function didn't update with this information, so provenance was messed up because the forwarding function's final proc was different form the actual final proc.

Reply:
Ah, got it. Looks good to me